Realm/RLMManagedDictionary.mm

```diff
@@ -368,29 +368,16 @@ - (void)removeObjectsForKeys:(NSArray *)keyArray {
     RLMAccessorContext context(*_objectInfo);
     changeDictionary(self, [&] {
         for (id key in keyArray) {
-            try {
-                _backingCollection.erase(context.unbox<realm::StringData>(key));
-            }
-            catch (realm::KeyNotFound const&) {
-                continue;
-            }
+            _backingCollection.try_erase(context.unbox<realm::StringData>(key));
         }
     });
 }
 
 - (void)removeObjectForKey:(id)key {
-    try {
-        changeDictionary(self, ^{
-            RLMAccessorContext context(*_objectInfo);
-            _backingCollection.erase(context.unbox<realm::StringData>(key));
-        });
-    }
-    catch (realm::KeyNotFound const&) {
-        return;
-    }
-    catch (...) {
-        throwError(nil, nil);
-    }
+    changeDictionary(self, ^{
+        RLMAccessorContext context(*_objectInfo);
+        _backingCollection.try_erase(context.unbox<realm::StringData>(key));
```

Comment:
What is the consequence of swallowing this error?

Reply:
Trying to remove a key from a dictionary which doesn't have that key isn't an error to begin with.